lib/dialects/sqlite/query-generator.js

```diff
@@ -288,12 +288,25 @@ const QueryGenerator = {
     options = options || {};
     _.defaults(options, this.options);
 
+    if (options.truncate === true) {
+      // Truncate does not allow LIMIT and WHERE
+      return `DELETE FROM ${this.quoteTable(tableName)}`;
+    }
+
     let whereClause = this.getWhereConditions(where, null, model, options);
     if (whereClause) {
       whereClause = ' WHERE ' + whereClause;
     }
 
-    return `DELETE FROM ${this.quoteTable(tableName)}${whereClause}`;
+    if (_.isUndefined(options.limit)) {
+      options.limit = 1;
+    }
+
+    if (options.limit) {
+      return `DELETE FROM ${this.quoteTable(tableName)} WHERE rowid IN (SELECT rowid FROM ${this.quoteTable(tableName)}${whereClause} LIMIT ${this.escape(options.limit)})`;
+    } else {
+      return `DELETE FROM ${this.quoteTable(tableName)}${whereClause}`;
```

Comment:
This case will never run, if limit is undefined you are setting is to 1, which will always trigger if case.

I suggest just remove `options.limit/isUndefined` part, switch query only when limit is supplied by user

Reply:
OK. Although I am still curious on why we would need the `options.limit/isUndefined` part from [`dialect/mysql.js`](https://github.com/sequelize/sequelize/blob/master/lib/dialects/mysql/query-generator.js#L182-L190), I guess we really should drop the conditional here. Updating it.

Reply:
That said, is there a case where `limit` is `0`, though? If so that would be a case that triggers the `else` clause?

Reply:
In that case it would be better to check for `isDefined` in if expression, because we want to limit DELETE even if its set to 0.

Reply:
> In that case it would be better to check for `isDefined` in if expression, because we want to limit DELETE even if its set to 0.

Do you want that to reflect on other dialects as well? Because I am afraid this is not how the current implementation for all other dialects is. That includes [MySQL](https://github.com/sequelize/sequelize/blob/17b19d9c67a3b6619f2cce4e920bc52b2b0fc4ab/lib/dialects/mssql/query-generator.js#L464), [PostgreSQL](https://github.com/sequelize/sequelize/blob/17b19d9c67a3b6619f2cce4e920bc52b2b0fc4ab/lib/dialects/postgres/query-generator.js#L410) and [MSSQL](https://github.com/sequelize/sequelize/blob/17b19d9c67a3b6619f2cce4e920bc52b2b0fc4ab/lib/dialects/mysql/query-generator.js#L188).